### ISSUE TITLE
[Bug] `IotaDID::check_method_id`does not error on invalid network names. 

### ISSUE THREAD

**Issue by olivereanderson** [iotaledger/identity.rs#957]
## Bug description
The `IotaDID::check_method_id` method allows network names of arbitrary length despite the  [DID Format section](https://wiki.iota.org/identity.rs/specs/did/iota_did_method_spec#did-format) of the [IOTA DID method specification](https://wiki.iota.org/identity.rs/specs/did/iota_did_method_spe) requiring the network name to be **at most six characters long.**  

## Language version
What programming language are you using? Rust or one of the bindings?
Which version of the language are you running?

- Language: Rust
- Version: 1.62.1

## IOTA Identity version
0.6 

## Steps To reproduce the bug
Place the following code in the test module of `identity_iota_core/did/iota_did.rs` 
```
  #[test]
  fn test_long_networkname() {
    let did: CoreDID = CoreDID::parse("did:iota:longnetworkname:H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV").unwrap();
    assert!(IotaDID::check_method_id(&did).is_err());
  }
``` 
then run `cargo test. 

## Expected behaviour
The test should pass. 

## Actual behaviour
The test fails because of the assertion not being valid. 

## Errors
```
---- did::iota_did::tests::test_long_networkname stdout ----
thread 'did::iota_did::tests::test_long_networkname' panicked at 'assertion failed: IotaDID::check_method_id(&did).is_err()', identity_iota_core/src/did/iota_did.rs:358:5
```

**Comment by PhilippGackstatter** [comment]
I don't think this is a bug. The network is not the method id, so they are checked independently. The network name length is checked in `IotaDID::check_network` and not in `IotaDID::check_method_id`.

**Comment by olivereanderson** [comment]
By `method_id` I assume one means `method-specific-id` per the [DID Core spec](https://www.w3.org/TR/did-core/#did-syntax), which in the case of `IOTA DID` should be `<network-name>:<tag>`.

**Comment by PhilippGackstatter** [comment]
I see, you're right. We should indeed update our code to match the spec.
